docs/modules/storage/pages/configuring-persistence.adoc

```diff
@@ -0,0 +1,242 @@
+scheduling, more contention on I/O, and less efficient garbage collection.
+
+=== `persistence.validation-timeout-seconds`
+
+Validation timeout for the Persistence process
+when validating the cluster members expected to join and the partition table on the whole cluster.
+
+=== `persistence.data-load-timeout-seconds`
+
+Data load timeout for the Persistence process.
+All members in the cluster should finish restoring their local data before this timeout.
+
+=== `persistence.cluster-data-recovery-policy`
+
+Specifies the data recovery policy that is
+respected during the Persistence cluster start.
+
+Default: `FULL_RECOVERY_ONLY`
+
+Valid values are:
+
+* `FULL_RECOVERY_ONLY`: Starts the cluster only when all expected members
+are present and correct. Otherwise, it fails.
+* `PARTIAL_RECOVERY_MOST_RECENT`: Starts the cluster with the members which have most up-to-date partition table and successfully restored their data. All other members leave the cluster and force start themselves. If no member restores its data successfully, cluster start fails.
+* `PARTIAL_RECOVERY_MOST_COMPLETE`: Starts the cluster with the largest group of members which have the same partition table version and successfully restored their data. All other members leave the cluster and force start themselves. If no member restores its data successfully, cluster start fails.
+
+=== `persistence.auto-remove-stale-data`
+
+Enables automatic removal of stale persistence stores.
+When a member terminates or crashes and the cluster state is `ACTIVE`, the remaining
+members redistribute the data among themselves and the data persisted on terminated
+member's storage becomes stale. That terminated member cannot rejoin the cluster
+without removing its persistence store. When auto-removal of stale data is enabled,
+while restarting that member, the persistence store is automatically removed and it joins
+the cluster as a completely new member. Otherwise, the persistence store should be removed manually.
+
+=== `persistence.encryption-at-rest`
+
+Configures encryption of data in the persistence store.
+See xref:encryption-at-rest.adoc[] for more information.
+
+Default: disabled
+
+== Data Structure Persistence Configuration
+
+Use the following options to configure Persistence for map and JCache data structures.
+
+=== `persistence.enabled`
+
+Whether the
+Persistence feature is enabled for the related data structure.
+
+Default: `false`
+
+=== `persistence.fsync`
+
+Turning on `fsync` guarantees that data is persisted to the disk
+device when a write operation returns successful response to the caller.
+
+Default: `false`
+
+By default, data is persisted to the disk device
+eventually, instead of on every disk write. This generally provides a better performance.
+
+=== `merkle-tree`
+
+Configures the Merkle tree for faster synchronization of data in the persistence store with the rest of the cluster.
```

Comment:
While this option is not strictly related to persistence (`IMap` merkle trees are also used for WAN sync), it is strongly recommended to enable merkle tree for each persistent `IMap` / `Cache`.

Reply:
Other than WAN sync, are there any other scenarios where map or cache Merkle trees are beneficial?

Reply:
If we have Merkle trees for JCache, can we remove this note from the delta WAN sync page?

```
Currently, Delta WAN Synchronization is implemented only for Hazelcast IMap. It will be implemented also for ICache in the future releases.
```

Reply:
We list these prerequisites for using Merkle trees with WAN, do they also apply to Persistence?

- Members/clusters must have the same number of partitions.
- Members/clusters must use the same partitioning strategy.
- Members/clusters must have the Merkle tree structure enabled.

Reply:
> Other than WAN sync, are there any other scenarios where map or cache Merkle trees are beneficial?

No, just WAN sync and only for `IMap`. Even though we added Merkle tree support for `ICache` in 5.0, we still lack the integration with WAN sync. So to sum up, merkle trees, as of 5.0, are useful for (1) IMap wan sync and differential partition sync, (2) ICache differential partition sync. Just FYI (2) will be done in 5.0 GA, it's not part of beta 1 or 2.

Reply:
About prerequisites, they do apply. However since for persistence we are discussing in the context of a single cluster, it is expected that all members have the same configuration eg you cannot have different partition count configured for members in the same cluster.